### DISCUSSION TITLE
Highspy - support assigning vartypes

### DISCUSSION THREAD

**Discussion by FabianHofmann** [ERGO-Code/HiGHS#971] [Ideas]
Thank you for setting up the great highspy package, it works like charme! I was wondering how to add vartype to the model via the python interface.  The `addVars` option takes `lower` and `upper` bounds but no vartype specification. Am I missing something?

Thank you in advance!

**Comment by jajhall** [comment]
Interesting observation, but the lack of a `HighsVarType` parameter is deliberate, as it's also missing from `addCol`. The variable type is defined using `highs::changeColIntegrality`. 

Perhaps we could have `Highs::addVar(const double lower, const double upper, const HighsVarType type=HighsVarType::kContinuous)` so that folk defining LPs don't have to specify HighsVarType::kContinuous, but then we'd have to have different methods in the C API to cover each member of `HighsVarType` other than `kContinuous`. 

Are you OK to call `highs::changeColIntegrality`?

**Comment by FabianHofmann** [comment]
@jajhall makes sense! `changeColsIntegrality` does the job  as it is fast and allows to assign multiple type at ones. Thanks!
